.github/workflows/automation-trigger-test.yml

```diff
@@ -1,18 +1,117 @@
+          CERTIFICATE_PATH=$RUNNER_TEMP/build_certificate.p12
+          KEYCHAIN_PATH=$RUNNER_TEMP/app-signing.keychain-db
+          PP_PATH=$RUNNER_TEMP/build_pp.mobileprovision
+
+          # import certificate 
+          echo -n "$BUILD_CERTIFICATE_BASE64" | base64 --decode -o $CERTIFICATE_PATH
+          echo -n "$BUILD_PROVISION_PROFILE_BASE64" | base64 --decode -o $PP_PATH
+
+          # create temporary keychain
+          security create-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          security set-keychain-settings -lut 21600 $KEYCHAIN_PATH
+          security unlock-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+
+          # import certificate to keychain
+          security import $CERTIFICATE_PATH -P "$P12_PASSWORD" -A -t cert -f pkcs12 -k $KEYCHAIN_PATH
+          security list-keychain -d user -s $KEYCHAIN_PATH
+
+          # apply provisioning profile
+          mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+          cp $PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+          
+      - name: Download Branch SDK lib, build testbed app and create IPA File
+        uses: actions/download-artifact@v3
+      - name : Unzip and copy folder
+        run: |
+            unzip release_framework/Branch_static.zip -d ./Framework/
+            xcodebuild -workspace DeepLinkDemo.xcworkspace -scheme DeepLinkDemo  -sdk iphoneos archive -archivePath ./IPA/DeepLinkDemo.xcarchive
+            xcodebuild -exportArchive -archivePath ./IPA/DeepLinkDemo.xcarchive -exportOptionsPlist IPA/Info.plist -exportPath IPA/
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: DeepLinkDemo.ipa
+          path: |
+                ./IPA/DeepLinkDemo.ipa
+  build-And-Run-AutomationTestSuite:
+    runs-on: macos-latest
+    needs: [build-Automation-testbed-App]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: BranchMetrics/qentelli-saas-sdk-testing-automation
+          ref: dev-iOS
+          token: ${{ secrets.BRANCHLET_ACCESS_TOKEN_PUBLIC }}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: maven
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3.3.0
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+      - name : Unzip and copy folder
+        run: |
+            ls -la ./
+            cp -R DeepLinkDemo.ipa testdata/builds/iOS_TestbedApp_21-Jun-2022-1.ipa
+      - name: Build with Maven
+        id: build_maven
+        run: |
+           mvn clean test -DsuiteXMLFile=testng.xml -e -DInputFormat="json" -DInputFilePath="TestSelection_BrowserStack_iOS.json" -DAccessKey="${{ secrets.S3_ACCESS_KEY }}" -DSecretKey="${{ secrets.S3_SECRET_KEY }}" --log-file output.txt
+      - name: Print output
+        if: always()        # run this step even if previous step failed
```

Comment:
use `success() || failure() `instead of `always()` as always causes the step to always execute, and returns true, even when `canceled`.
Also, always will execute even when any of the steps above in the job fail meaning not just the success or failure of `build_maven` step but steps above it and in that case, you won’t have output.txt so its better to check
`if: success() || steps. build_maven.outcome == 'failure'`